### ISSUE TITLE
Different behaviour between `nbqa black` and `nbqa black --check`

### ISSUE THREAD

**Issue by tcbegley** [nbQA-dev/nbQA#626]
Hello,

Thanks for making this awesome tool.

I've been having some problems related to `nbqa black --check` raising problems but `nbqa black` not doing anything to change it. I've managed to recreate this with the following minimal example. It appears to be related to how the configuration is parsed.

```
.
├── notebook.ipynb
└── pyproject.toml
```

My project has something like the following configuration. We set the line length to be 79 for source code but are happy with longer lines inside notebooks.

```toml
# pyproject.toml
[tool.black]
line-length = 79

[tool.nbqa.addopts]
# use longer lines in notebooks
black = [
    "--line-length=88",
]
```

`notebook.ipynb` has a single cell with the following contents

```python
result = function_name(xxxxxxxxxxxxxxxxxxxxxxxxxx, xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx)
```

The exact contents are not so important, the important thing is that the line length is more than 79 but less than 88. Running

```sh
nbqa black notebook.ipynb
```

results in no changes, but running

```sh
nbqa black --check notebook.ipynb
```

indicates that the notebook would be reformatted.

![image](https://user-images.githubusercontent.com/15220906/126333704-1345f9de-90bd-4e0c-a8b6-6b6cc1fb5106.png)

To avoid any ambiguity, here's the notebook JSON

```json
{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77402ebb-99fc-4232-9f9a-d46bb0d3aa3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "result = function_name(xxxxxxxxxxxxxxxxxxxxxxxxxx, xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "nbqa",
   "language": "python",
   "name": "nbqa"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
```

**Comment by tcbegley** [comment]
It seems that the `addopts` specified in `pyproject.toml` get overridden [here](https://github.com/nbQA-dev/nbQA/blob/master/nbqa/__main__.py#L323-L327). Would it make sense to concatenate the lists (with command line arguments coming after so they override any `pyproject.toml` options) so that both get applied?

**Comment by MarcoGorelli** [comment]
Hey @tcbegley !

> Thanks for making this awesome tool.

Thanks! Are you using it at Faculty AI?

---

> It seems that the `addopts` specified in `pyproject.toml` get overridden [here](https://github.com/nbQA-dev/nbQA/blob/master/nbqa/__main__.py#L323-L327).

Yes, that's right. Thanks for investigating this and for sending across such an easy-to-reproduce report!

> Would it make sense to concatenate the lists (with command line arguments coming after so they override any `pyproject.toml` options) so that both get applied?

Is it always the case that command-line arguments which come after override the earlier ones? If so, then that sounds like a good suggestion, as that was my reservation around combining the lists - do you want to make a PR?

---

As an aside, I'm working on getting a PR into `black` so that Jupyter Notebooks will be supported directly - https://github.com/psf/black/pull/2357 . If you have lots of notebooks you're running `black` on, it'd be really useful if you could try checking out that branch, running `black` on them, and report on any bugs or anything unexpected

**Comment by tcbegley** [comment]
> Thanks! Are you using it at Faculty AI?

I am personally and I'm doing my best to evangelise 😛 

> Is it always the case that command-line arguments which come after override the earlier ones?

I did a quick test on `black` and `black --line-length=88 --line-length=79` etc. has exactly this behaviour. I don't feel like I know enough to be sure whether it's safe to rely on it always working? Looking at the [ArgParse docs](https://docs.python.org/3/library/argparse.html#action) it seems with e.g. `action="append"` you can collect repeated optional arguments rather than having them overwrite each other. Not sure if any of the tools nbQA supports actually use this behaviour? Even if they do there might actually be use cases where the user wants to supply arguments that are _additional_ to those specified in `pyproject.toml`?

Basically I'm not sure! I'd be happy to prepare a PR to play around with it and do some testing?

> As an aside, I'm working on getting a PR into black so that Jupyter Notebooks will be supported directly

Sounds awesome! Will definitely check it out.

**Comment by MarcoGorelli** [comment]
>  Not sure if any of the tools nbQA supports actually use this behaviour? Even if they do there might actually be use cases where the user wants to supply arguments that are additional to those specified in pyproject.toml?

In theory, nbQA should be able to run any Python tool, even a local script. So yes, this would create issues in the case of tools/scripts which have an argument with action "append" if a user wanted to override what was in their pyproject.toml file. _However_, such cases strike me as exceedingly rare - certainly rarer than the use-case you've presented here.

So I'm inclined to go ahead with combining the lists for `--nbqa-addopts` and to clearly document that for this flag, the options are combined with those in pyproject.toml rather than overriding them.

And if someone has some use case for which have a tool with a flag with action "append" for which they want to override pyproject.toml options with command-line ones, I expect they'll open an issue and I can think about supporting that

**Comment by MarcoGorelli** [comment]
> Sounds awesome! Will definitely check it out.

Thanks - here's how to install it:
```
pip uninstall black -y
pip install git+https://github.com/MarcoGorelli/black.git@jupyter
pip install -U IPython tokenize-rt typing-extensions
```

**Comment by MarcoGorelli** [comment]
closed in #627

**Comment by MarcoGorelli** [comment]
@tcbegley I've made a new release (it's been about a year since the first version, so I've finally made a 1.0.0 release :tada:), if you could try it out and let me know if this doesn't work for you, that'd be much appreciated

As an aside, `--nbqa-mutate` is no longer necessary (it won't error if you pass it, but it won't do anything anymore)

**Comment by tcbegley** [comment]
Thanks @MarcoGorelli! Updated our CI and it all seems to be working like a charm.
